src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Unix.cs

```diff
@@ -290,5 +291,21 @@ private static void OnAvailabilityTimerFired(object? state)
                 }
             }
         }
+
+        // SocketWrapper class was introduced to prevent event thread leaks on Linux
+        // as described on GitHub: https://github.com/dotnet/runtime/issues/63788
+        // For each CreateSocket() new s_socket instance is created and we pass it to ".NET Network Address Change" thread.
+        // We then continue loop in LoopReadSocket method as long as both references are equal.
+        private class SocketWrapper
+        {
+            public SocketWrapper(int socket)
+            {
+                Socket = socket;
+            }
+
+            public int Socket { get; }
+
+            public static readonly SocketWrapper NotSet = new SocketWrapper(0);
```

Comment:
While unlikely, 0 is valid handle. We should perhaps use -1

Reply:
Hi, thank you for this!

With commit https://github.com/dotnet/runtime/pull/63963/commits/b6e944d44b40ec6e68b3b69b73021fd538558ffd this is now changed, we simply set `SocketWrapper` to null as suggested by @tmds